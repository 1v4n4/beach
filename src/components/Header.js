import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import {
  Navbar, Nav, Container, Button,
} from 'react-bootstrap';
import { useSelector, useDispatch } from 'react-redux';
import { setUser, getUser } from '../localStorage';
import { logoutUser } from '../actions/logOutActions';
import { checkUser } from '../actions/userActions';
import Logged from './Logged';
import '../CSS/header.css';

const Header = () => {
  const dispatch = useDispatch();

  const result = getUser() || {};
  const { id } = result;
  const FetchUser = () => {
    if (result.id) {
      dispatch(checkUser(result));
    }
  };

  useEffect(() => {
    FetchUser();
  }, []);
  const user = useSelector((state) => state.user);
  const isLogged = user.logged;

  const handleClick = () => {
    setUser(null);
    dispatch(logoutUser());
  };

  return (
    <Navbar bg="light" expand="lg">
      <Container>
        <Navbar.Brand to="#home"><Link className="text-dark fs-3" to="/">Home</Link></Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            { !isLogged
          && <Link className="nav-link text-center bg-light log" to="/login">Log In</Link> }
            { !isLogged && <Link className="nav-link text-center bg-light log" to="/signup">Sign Up</Link> }
            { isLogged
          && <Logged id={id} /> }
            { isLogged && <Button className="nav-link bg-light log" onClick={handleClick}>Log Out</Button> }

          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};
export default Header;
