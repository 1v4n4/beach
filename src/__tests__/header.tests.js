import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';
import createTestStore from './store';

describe('Header component', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });

  test('should match with snapshot', () => {
    const jsx = (
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
    const tree = (() => renderer.create(jsx).toJSON());
    expect(tree).toMatchSnapshot();
  });
});
