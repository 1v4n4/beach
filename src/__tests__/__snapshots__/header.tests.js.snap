// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match with snapshot 1`] = `[Function]`;
